### ISSUE TITLE
options command generating errors about missing ports tree FLAVOR feature

### ISSUE THREAD

**Issue by madpilot78** [freebsd/poudriere#1083]
## Prerequisites
- [x] Have you checked for an existing issue describing your problem?
- [x] Are you running the latest version?
- [x] Is your ports tree recent?
- [x] Is your FreeBSD Host on a [supported release](https://www.freebsd.org/security/#sup)?

## Describe the bug

With latest `ports-mgmt/poudriere-devel` port running the `poudriere options` command passinga  port with a flavor (f.e. `net/unison@nox11`) causes errors like:

```
[00:00:00] Error: Trying to build FLAVOR-specific net/unison@nox11 but ports tree has no FLAVORS support.
```

And the port is skipped.

## How to reproduce

see above

## Expected behavior
No errors being output. No skipping

## Environment
 - Host OS [e.g. 12.2 amd64]: head
 - Jail OS [e.g. 12.0 powerpc]: any
 - Poudriere Version [e.g. 3.3.1 or git hash or port version]: 3.4.99.20231030
 - Ports branch and revision [e.g. 2020Q3 r550754]: head

## Additional context

After further investigation, when running the options command, the P_PORTS_FEATURES variable is not being populated via the options command.

If I get more insight to share I'll do.

**Comment by madpilot78** [comment]
The SUBPACKAGES feature is still being worked on, support needs to be added in poudriere before it can be unleashed on the ports tree.

But we should be getting it at some point.

Looks like the bug was introduced when adding support for SUBPACKAGES and not updating the options.sh script, but looking at it I found nothing obvious about how to fix this.

**Comment by madpilot78** [comment]
> * Handle missing ports features gracefully.

This is in theory already done. Obviously if you ask for something that depends on a feature (like a specific flavor) there is not much to be done, skipping it is correct. Reverting on the unflavored port looks arbitrary.

Anyway the only issue at hand is the options subcommand not properly populating `$P_PORTS_FEATURES`.

I tried to figure it out but could not find a solution. I plan to find some spare time in the near future to try harder.

**Comment by madpilot78** [comment]
I have created a pull request with a patch that works fine for me.

You can test it and report back.

Find it here: #1087

Thanks!

**Comment by ahktenzero** [comment]
I updated poudriere-devel to 3.4.99.20231113 which includes the changes in #1087 and the options command now fails with an error saying the port does not exist for any port I try. It's looking for the port in `/usr/ports` (which is empty on my system) and not the ports tree I use for building packages at `/usr/local/poudriere/ports/default`. 

After adding some debug output, the problem is that `jail_start` sets `PORTSDIR` to `/usr/ports`, which is correct for commands running inside the jail as the ports tree gets mounted there, but in `options.sh` the commands to check and update the options are run outside the jail so it ends up looking at the wrong ports tree.

I've uploaded a PR with a fix, #1089.

**Comment by madpilot78** [comment]
@ahktenzero Thanks for noticing my mistake, I clearly missed this detail of jail_start.

**Comment by bapt** [comment]
https://bapt.nours.eu/patch-fix-options-for-flavors.diff this should fix this without having to spawn a jail by default

**Comment by madpilot78** [comment]
> https://bapt.nours.eu/patch-fix-options-for-flavors.diff this should fix this without having to spawn a jail by default

@bapt I've tested it with my common usage patterns and it looks fine, does what is expected.

**Comment by madpilot78** [comment]
Fixed by 9143ab494773f7a1ba69efc31177d4bbfe51bf7f
